crates/sui-graphql-rpc/schema.graphql

```diff
@@ -0,0 +1,823 @@
+# Copyright (c) Mysten Labs, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+# Open Questions
+# --------------
+# - Rules for nullabilit/non-nullability
+# - Pattern for multi-gets
+# - Encoding RPC errors
+# - `type` (vs `scalar`) for Move Types
+# - Pattern for upcasting and downcasting
+
+# Check Completeness Against Existing APIs
+# -------------
+# DONE Read
+# DONE - Objects
+# DONE - Transactions
+# DONE - Checkpoints
+# DONE - Events
+# DONE - ProtocolConfig
+# DONE - Chain Identifier
+# DONE Coin
+# DONE Extended
+# DONE Governance
+# DONE Indexer
+# DONE Move Utils
+# TODO Write
+
+# DONE Port designs from Notion
+# DONE - Display
+# DONE - SuiNS
+# DONE Data Consistency API
+# TODO Flesh out Connections
+# TODO - TotalTransactionBlocks for the PTB connection
+
+schema {
+  query: Query
+  subscription: Subscription
+}
+
+type Query {
+  # First four bytes of the network's genesis checkpoint digest
+  # (uniquely defines the network)
+  chainIdentifier: String!
+
+  # First and last checkpoint that the RPC can answer consistent
+  # queries for.
+  consistentRange: ConsistentRange
```

Comment:
`consistent` may not a be the best word, because the node/rpc provider could be continuously running pruning and keeping up to date with the network hence moving this range. I think a more conservative name is `AvailableRange` or `CurrentlyAvailableRange`

Reply:
I guess the issue with "available" as a term is that especially from pruning, there may be older checkpoints that the RPC can answer queries for on some endpoints but not others, so we can't say that all checkpoints outside this range are "unavailable".  Agreed though that `consistent` is also not great.